cli/src/commands/operation/undo.rs

```diff
@@ -70,5 +71,25 @@ pub fn cmd_op_undo(
     tx.repo_mut().set_view(new_view);
     tx.finish(ui, format!("undo operation {}", bad_op.id().hex()))?;
 
+    if let Some(mut formatter) = ui.status_formatter() {
+        writeln!(formatter, "Undid operation {}", bad_op.id().hex())?;
+        let workspace_env = workspace_command.env();
+        let language = OperationTemplateLanguage::new(
+            bad_op.op_store().root_operation_id(),
+            Some(bad_op.id()),
+            workspace_env.operation_template_extensions(),
+        );
+        let text = command
+            .settings()
+            .config()
+            .get_string("template-aliases.builtin_op_undo")?;
```

Comment:
nit: `[templates]`, not `[template-aliases]`

Reply:
Why do the two exists? Is the former newer?

Reply:
The thinking was that `[template-aliases]` is an open namespace where the user can define whatever alias they want but `[templates]` has a predefined set of keys, which means we can introduce a new key there later without having to worry that the user has already defined an alias with that name. However, we now the the `builtin-` aliases that are also predefined and implicitly not for the user to add to. So I think we have said that we could remove the `[templates]` section. Right, @yuja?

Reply:
Since user shouldn't (conceptually) override `builtin_*` aliases, we'll need `<name> = builtin_<name>` indirection somewhere. That's `templates.<name> = builtin_<name>_<style>` right now.

iirc, `[templates]`/`[template-aliases]` distinction isn't super important. It's just copied from revsets where symbol aliases shadow tag/branch names, so no builtin symbol aliases should be added to `[revset-aliases]`.